src/components/game/coreState/CoreState.js

```diff
@@ -289,14 +322,29 @@ const CoreState = class {
   // Change the CoreState's grid values based on where the current piece is.
   placeCurrentPiece() {
     if (this.currPiece != null) {
-      var [x, y] = [0, 0];
-      for (const cell of this.currPiece.cells) {
-        [x, y] = [
-          cell[1][0] + this.currPiece.cx,
-          cell[1][1] + this.currPiece.cy,
-        ];
-        if (x >= 0 && x < this.boardSize && y >= 0 && y < this.boardSize) {
-          this.board[y][x] = this.currPiece.createCell();
+      if (this.currPiece.mainCell.type == 3) {
+        for (var y = Math.max(0, this.currPiece.cy - 2); y < Math.min(this.boardSize, this.currPiece.cy + 3); y++) {
+          for (var x = Math.max(0, this.currPiece.cx - 2); x < Math.min(this.boardSize, this.currPiece.cx + 3); x++) {
+            this.board[y][x] = this.emptyValue();
+          }
+        }
+      } else if (this.currPiece.mainCell.type == 4) {
+        dropzone(this.board, this.currPiece, this.gravity, (cell) => {
+          cell.type = 0;
+          cell.props = {...DEFAULT_CELL_PROPS}
+        }, true)
+      } else {
+        var [x, y] = [0, 0];
+        for (const cell of this.currPiece.cells) {
+          [x, y] = [
+            cell[1][0] + this.currPiece.cx,
+            cell[1][1] + this.currPiece.cy,
+          ];
+          if (x >= 0 && x < this.boardSize && y >= 0 && y < this.boardSize) {
+            this.board[y][x] = new Cell(1,
```

Comment:
getAttributesFrom is a good way of doing this. Probably cannot one-line it since this is used for OOP-level swapping where we still need the original reference, and given all the new Cell subclasses.